packages/one-app-bundler/webpack/loaders/validate-required-externals-loader.js

```diff
@@ -12,30 +12,47 @@
  * under the License.
  */
 
+const fs = require('node:fs');
+const path = require('node:path');
 const loaderUtils = require('loader-utils');
 const readPkgUp = require('read-pkg-up');
 
 function validateRequiredExternalsLoader(content) {
   const options = loaderUtils.getOptions(this);
   const { packageJson } = readPkgUp.sync();
+  const integrityManifest = JSON.parse(fs.readFileSync(path.join(process.cwd(), 'bundle.integrity.manifest.json'), 'utf-8'));
+  const requiredExternals = options.requiredExternals.reduce((obj, externalName) => {
+    // eslint-disable-next-line global-require, import/no-dynamic-require -- need to require a package.json at runtime
+    const { version } = require(`${externalName}/package.json`);
+    const semanticRange = packageJson.dependencies[externalName];
 
-  const requiredExternals = options.requiredExternals.map((externalName) => {
-    const version = packageJson.dependencies[externalName];
-    return `'${externalName}': '${version}'`;
-  });
+    return {
+      ...obj,
+      [externalName]: {
+        name: externalName,
+        version,
+        semanticRange,
+        integrity: integrityManifest[externalName],
+      },
+    };
+  }, {});
   const match = content.match(/export\s+default\s+(?!from)(\w+);$/m);
 
   if (match) {
     const newContent = `${content};
 if (!global.BROWSER) {
   ${match[1]}.appConfig = Object.assign({}, ${match[1]}.appConfig, {
-    requiredExternals: {
-      ${requiredExternals.join(',\n      ')},
-    },
+    requiredExternals: ${JSON.stringify(requiredExternals, null, 2)},
```

Comment:
this needs to be updated to retain the old api on app config, new api is being consumed from module-config.